rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/toolWindow/log/UnityLogPanelEventRenderer.kt

```diff
@@ -4,17 +4,217 @@ import com.intellij.ui.ColoredListCellRenderer
+            }
+        }
+
+        tokens.add(Token(fullString.substring(lastTokenIndex), TokenType.String))
+
+        return tokens
+    }
+
+    private fun addTokens(i: Int, lastTokenIndex: Int, tokens: MutableList<Token>, fullString: String, tokenString: String, type: TokenType) {
+        if (i > lastTokenIndex)
+            tokens.add(Token(fullString.substring(lastTokenIndex, i), TokenType.String))
+        tokens.add(Token(tokenString, type))
+    }
+
+    private fun getToken(fullString: String, startIndex: Int) : String
+    {
+        if(fullString[startIndex] != '<')
+            return ""
+
+        for (i in startIndex until fullString.length) {
+            if(fullString[i] == '>')
+                return fullString.substring(startIndex until i)
+        }
+
+        return ""
+    }
+
+    private fun checkToken(fullString: String, expectedToken: String, startIndex: Int): Int {
+        for (i in startIndex until fullString.length) {
+            val expectedChar = expectedToken[i - startIndex]
+
+            if (fullString[i] != expectedChar) {
+                return -1
+            }
+
+            if (i - startIndex == expectedToken.length - 1) {
+                return i
+            }
+        }
+
+        return -1
+    }
+
+    private fun parseColor(color: String): Color? {
+        try {
+            when (color) {
+                "aqua" -> return Color.decode("#00ffff")
+                "black" -> return Color.decode("#000000")
+                "blue" -> return Color.decode("#0000ff")
+                "brown" -> return Color.decode("#a52a2a")
+                "cyan" -> return Color.decode("#00ffff")
+                "darkblue" -> return Color.decode("#0000a0")
+                "fuchsia" -> return Color.decode("#ff00ff")
+                "green" -> return Color.decode("#008000")
+                "grey" -> return Color.decode("#808080")
+                "lightblue" -> return Color.decode("#add8e6")
+                "lime" -> return Color.decode("#00ff00")
+                "magenta" -> return Color.decode("#ff00ff")
+                "maroon" -> return Color.decode("#800000")
+                "navy" -> return Color.decode("#000080")
+                "olive" -> return Color.decode("#808000")
+                "orange" -> return Color.decode("#ffa500")
+                "purple" -> return Color.decode("#800080")
+                "red" -> return Color.decode("#ff0000")
+                "silver" -> return Color.decode("#c0c0c0")
+                "teal" -> return Color.decode("#008080")
+                "white" -> return Color.decode("#ffffff")
+                "yellow" -> return Color.decode("#ffff00")
+                else -> return when {
+                    color.length == 8 -> Color.decode(color.substring(0..7))
+                    color.length == 7 -> Color.decode(color)
+                    else -> null
+                }
+            }
+        } catch (t: Throwable) {
+            return Color.white
```

Comment:
Unity shows white if it can't parse the colour properly, I guess they're trying to show if your colour string is incorrect. Do we want to do the same or fall back to the default colour?

Reply:
I think fallback to the default colour. If we in the light colour scheme, we wouldn't be able to see white text. If it's default, we'll see the text on both light and dark schemes.